src/Discord.Net.Rest/Net/Queue/GatewayBucket.cs

```diff
@@ -0,0 +1,62 @@
+using Discord.Rest;
+using System.Collections.Immutable;
+
+namespace Discord.Net.Queue
+{
+    public enum GatewayBucketType
```

Comment:
Should this be in its own class?

Reply:
I followed the same design as `ClientBucket` that has `ClientBucketType` in the same file, should that be changed?

Reply:
Requesting input from other contributors.
cc @foxbot @Joe4evr

Reply:
Idk about this, so I defer to fox's judgement.